src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -341,33 +313,36 @@ private void _ReadResources()
             {
                 throw new BadImageFormatException(SR.BadImageFormat_ResourcesHeaderCorrupted);
             }
-
-
         }
        
-        private Type FindType(int typeIndex)
+        private bool IsString(int typeIndex)
         {
-            if (typeIndex < 0 || typeIndex >= _typeTable.Length)
+            if (typeIndex < 0 || typeIndex >= _numOfTypes)
             {
                 throw new BadImageFormatException(SR.BadImageFormat_InvalidType);
             }
-            if (_typeTable[typeIndex] == null)
-            {
-                long oldPos = _stream.Position;
-                try
-                {
-                    _stream.Position = _typeNamePositions[typeIndex];
-                    string typeName = _stream.ReadString();
-                    _typeTable[typeIndex] = Type.GetType(typeName, true);
-                }
 
-                finally
-                {
-                    _stream.Position = oldPos;
-                }
+            return typeIndex == _stringTypeIndex;
+        }
+
+        private static byte[] EncodesStringName()
+        {
+            var type = typeof(string);
+            var name = type.AssemblyQualifiedName;
+            int length = name.IndexOf(", Version=");
+
+            var buffer = new byte[length + 1];
```

Comment:
I assume the assembly-qualified name will always contain the version string such that we don't need to worry about length being -1?